content/articles/bigdecimal-and-primitive-data-type-arithmetic-program-in-java/index.md

```diff
@@ -0,0 +1,212 @@
+         // Adding both of them
+        System.out.println("Adding first1 and sec2 ="
+                           + first1.add(sec2));
+    }
+}
+```
+
+> Upon running the code above, there will be an error that will happen to state that `int` cannot be converted to `BigDecimal`.
+
+The error that occurs happens since the operation can only be performed on BigDecimal objects, and this is the case. Our primitive data type must be converted into a BigDecimal object using the BigDecimal class's constructor. 
+
+We are constructing a new object of class BigDecimal with the same value as `sec2` and providing it directly to the `add()` method as an argument. The code below resolves the problem in the following way:
+
+```Java
+import java.io.*;
+import java.math.BigDecimal;
+class Example {
+    public static void main(String[] args)
+    {
+
+        // Declaring a BigDecimal object and integer number
+        BigDecimal first1 = new BigDecimal("15");
+        int sec2 = 20;
+        System.out.println(
+            "Adding first1 and sec2 = "
+            + first1.add(new BigDecimal(sec2)));
+    }
+}
+```
+
+Output:
+
+```
+Adding first1 and sec2 = 35
+```
+
+Lastly, look at the following example showing the arithmetic operations between `BigDecimal` and `int` to understand it better:
+
+```Java
+import java.io.*;
+import java.math.BigDecimal;
+class Example {
+    public static void main(String[] args)
+    {
+
+        // Declaring an integer number and a BigDecimal object
+        BigDecimal first1 = new BigDecimal("15");
+        int sec2 = 20;
+        System.out.println(
+            "Adding first1 and sec2 = "
+            + first1.add(new BigDecimal(sec2)));
+
+        System.out.println(
+            "Subtracting first1 and sec2 = "
+            + first1.subtract(new BigDecimal(sec2)));
+
+        System.out.println(
+            "Multiplying first1 and sec2 = "
+            + first1.multiply(new BigDecimal(sec2)));
+
+        System.out.println(
+            "Dividing first1 and sec2 = "
+            + first1.divide(new BigDecimal(sec2)));
+    }
+}
+```
+
+Output:
+
+```
+Adding first1 and sec2 = 35
+Subtracting first1 and sec2 = -5
+Multiplying first1 and sec2 = 300
+Dividing first1 and sec2 = 0.75
+```
+
+> No null values are allowed in a database, application, or view and everything is initialized with new BigDecimal(0), or you execute null checks on every use of nullable values.
+
+### Need for BigDecimal
+There is a lot of fun to be had with floating-point numbers. A double type is commonly used for quantities unless the value is an integer, in which case an int type is usually acceptable. 
+
+Also, a float or a long can be used, depending on the size of a value's value type. It should be noted that these kinds are the very worst thing you can use when dealing with money. They do not provide the correct value, but rather one that can be stored in a binary format. As an example, let's look at the dangers of utilizing a double.

```

Comment:
On the last sentence of this paragraph, there is mention of an example but you haven't given any.